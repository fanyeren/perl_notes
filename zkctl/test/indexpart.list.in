#!/bin/bash
source $(dirname $0)/common

TYPE=index-partition

node_clear $TYPE
node_clear index-type

#没有index-type时添加失败
echo "Add with no index-type"
./zkctl -w $TYPE add --$TYPE-id=dnews_0 --data={} 2>&1
echo "Add with no index-type failed: result=$?"
#准备数据, index-type
./zkctl -w index-type add --index-type-id=dnews --data={}
./zkctl -w index-type add --index-type-id=wp --data={}
echo "add index type succedd result=$?"
./zkctl -w index-type add --index-type-id=vip --data={}
./zkctl -w index-type add --index-type-id=se --data={}
#正常添加
./zkctl -w $TYPE add --$TYPE-id=dnews_0 --data={}
echo "Add with right index-type succeed: result=$?"
./zkctl -w $TYPE add --$TYPE-id=wp_2 --data={}
./zkctl -w $TYPE add --$TYPE-id=vip_1 --data={}
./zkctl -w $TYPE add --$TYPE-id=se_3 --data={}
./zkctl -w $TYPE list | sort

#删除indextype
./zkctl -w index-type del --index-type-id=wp 2>&1
echo "delete should failed. result=$?"
#删除
./zkctl -w $TYPE del --$TYPE-id=wp_2 
echo "delete index partition. result=$?"
./zkctl -w $TYPE list | sort
#删除已经删除的
#./zkctl -w $TYPE del --$TYPE-id=wp_2 --data={} 2>&1
echo "Delete already deleted result=$?"

node_clear $TYPE
node_clear index-type
