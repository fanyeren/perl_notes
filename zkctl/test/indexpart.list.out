Add with no index-type
index-partition dnews_0 add failed, it may be exist already
Add with no index-type failed: result=1
add index type succedd result=0
Add with right index-type succeed: result=0
dnews_0
se_3
vip_1
wp_2
index-type del failed, probally has child
delete should failed. result=1
delete index partition. result=0
dnews_0
se_3
vip_1
Delete already deleted result=0
